File: Examples/VisualBasic/WorkingWithResources/GetResourceWorkVariance.vb

Namespace WorkingWithResources
    Public Class GetResourceWorkVariance
        Public Shared Sub Run()
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName)

            ' ExStart:GetResourceWorkVariance
            ' Read the input Project file
            Dim project As New Project(dataDir & "WorkVariance.mpp")

            For Each resourceAssignement1 As ResourceAssignment In project.ResourceAssignments
                Dim resource1 As Resource = resourceAssignement1.Get(Asn.Resource)
                Dim workVariance As Double = resource1.Get(Rsc.WorkVariance)
                Console.WriteLine(workVariance)
            Next
            ' ExEnd:GetResourceWorkVariance
        End Sub
    End Class
End Namespace

File: Examples/CSharp/WorkingWithResources/GetResourceWorkVariance.cs
namespace Aspose.Tasks.Examples.CSharp.WorkingWithResources
{
    using System;

    internal class GetResourceWorkVariance
    {
        public static void Run()
        {
            var dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod());

            //ExStart:GetResourceWorkVariance
            //ExFor: Rsc.WorkVariance
            //ExSummary: Shows how to read resource work variance.
            // Read the input Project file
            var project = new Project(dataDir + "WorkVariance.mpp");

            foreach (var assignment in project.ResourceAssignments)
            {
                var resource = assignment.Get(Asn.Resource);

                var workVariance = resource.Get(Rsc.WorkVariance);

                Console.WriteLine(workVariance);
            }
            //ExEnd:GetResourceWorkVariance
        }
    }
}
